### PR TITLE
Small performance related changes

### PR DESCRIPTION
**Pull request by vrmiguel**
We can always be moar blazingly fast

### CODE REVIEW

File: src/operations/generate.rs [hunk 1]
```diff
@@ -45,20 +45,16 @@ fn color_pass(
     min_saturation: Option<f32>,
     max_saturation: Option<f32>,
 ) -> Option<Rgb> {
-    let mut chosen = None;
-    for color in colors {
-        let (saturation, luma) = grab_sat_luma(*color);
-        if (max_luma == None || luma <= max_luma.unwrap())
+    let predicate = |rgb: &Rgb| {
```
Comment (vrmiguel) [inline src/operations/generate.rs:48]: I wish this closure had a better name but to be honest I have very little clue on what it does

**Review by Misterio77 — APPROVED** [review]
LGTM, thanks!

**Comment by igoracmelo** [comment]
LGTM

**Review by marcospb19 — COMMENTED** [review]
LGTM.

**Pull request merged**